### DISCUSSION TITLE
Possibility to remove the try-catch blocks from routes handler?

### DISCUSSION THREAD

**Discussion by thboss** [PhlexPlexico/G5API#154] [Ideas]
Hey @PhlexPlexico hope you doing fine,

I notice you added this error handler middleware
https://github.com/PhlexPlexico/G5API/blob/c8fa5fc7864925417a2fc502b829fa44474b088a/app.js#L188-L196
But it doesn't seem to be called since all routes are handled with try-catch blocks.
So, I would suggest using this package which allows removing try-catch block safely and make the code a little bit cleaner
https://www.npmjs.com/package/express-async-errors

**Comment by PhlexPlexico** [comment]
Hey! It's been going well, busy holidays and all!

As for that handler, I think that's because when I started out with the boilerplate template I had made it.

I think considering all routes are wrapped in try catch it certainly would be quite a rewrite, however I would like to avoid using too many packages to accomplish things (it's been an absolute terror to upgrade these in the past few days haha). I can definitely take a look in the new year here once I'm a bit more settled with personal matters though :) thanks for pointing this out!

**Reply by PhlexPlexico** [reply]
Also I think another reason is because I have some more custom messages in those try catch blocks instead of outputting errors that are bubbled up from SQL and whatnot, so it's a bit more concise and doesn't give away too much to an end user as well
